path/resolver/resolver.go

```diff
@@ -69,19 +69,28 @@ func (r *Resolver) ResolveToLastNode(ctx context.Context, fpath path.Path) (ipld
 	}
 
 	for len(p) > 0 {
-		val, rest, err := nd.Resolve(p)
-		if err != nil {
-			return nil, nil, err
-		}
+		lnk, rest, err := r.ResolveOnce(ctx, r.DAG, nd, p)
+		if lnk != nil {
```

Comment:
maybe we could clean this loop up a little bit by pulling everything below this if statement out of the for loop, then reversing the logic to 'break' if `lnk == nil`